### ISSUE TITLE
Adding Py3 support

### ISSUE THREAD

**Issue by fjarri** [bh107/bohrium#18]
This is not a bug, but rather a feature proposal. 

Currently the build uses Cheetah, the last version of which was released 5 years ago, and which does not support Py3. There are PR's attempting to fix that in [their github repo](https://github.com/cheetahtemplate/cheetah/pulls), but they do not seem to get much attention. 

I could prepare a PR switching to some more actively maintained templating engine ([Mako](http://www.makotemplates.org/) is my personal favourite, but I do not insist on it) and fixing possible other incompatibilities with Py3, but I would like to ask first if it is actually needed.

**Comment by safl** [comment]
Increased portability is very welcome :)

Cheetah is currently used for autogenerating a variety of things in the "core" of Bohrium, as well as the C++ bridge (Bxx), CIL bridge and the cpu-engine, this is just a fair warning that I might be a lengthy task.

If you choose to pursue it then there's really only two requirements/considerations:

1) The chosen template engine must be readily available on a most Linux distributions as well as MacOSX and Windows, and preferably a couple of versions/generations back.

2) Cheetah supports C/C++ code in templates, however, it is kind of messy since it requires stuff like this:

```#slurp
#compiler-settings
directiveStartToken = %
#end compiler-settings
```

In the templates. I would be an improvement to have "cleaner" templates. Perhaps a templating engine that is aware of the template code and the template-markup and thereby able to some sort of syntax high lightning, well that would be really great.

Ohhh... and thanks for your interest in Bohrium!

**Comment by madsbk** [comment]
Fixed in PR #142
